packages/cli/src/models/dependency/Dependency.ts

```diff
@@ -97,22 +97,18 @@ export default class Dependency {
       contractAlias,
     ]) as [Contract, string][];
 
-    const pipeline = [
-      someContracts => map(someContracts, ([contract]) => getSolidityLibNames(contract.schema.bytecode)),
-      someContracts => flatten(someContracts),
-      someContracts => uniq(someContracts),
-    ];
-
-    const libraryNames = pipeline.reduce((xs, f) => f(xs), contracts);
+    const libraryNames = uniq(flatten(contracts.map(([contract]) => getSolidityLibNames(contract.schema.bytecode))));
 
     const libraries = fromPairs(
       await Promise.all(
         map(libraryNames, async libraryName => {
-          const implementation = await project.setImplementation(
-            Contracts.getFromNodeModules(this.name, libraryName),
-            libraryName,
-          );
-          return [libraryName, implementation.address];
+          if (typeof libraryName === 'string') {
```

Comment:
Can `libraryName` be something that is not a string? 
If yes, I'd handle it explicitly.
If not, I'd remove this `if` clause and just type it.

Reply:
To my understanding `libraryName` is always a string. Its type is lost somewhere up at the callstack. I didn't have time to fix that. By just typing you mean `libraryName as string`? I think keeping is better because it will prevent errors if `libraryName` type will change due to code modifications when `libraryName as string` will overwrite any type information.

Reply:
I was thinking of typing `libraryNames` above, and also changing:
```diff
- map(libraryNames, async libraryName => {
+ map(libraryNames, async libraryName: string => {
```